code/modules/wiremod/components/atom/remotecam.dm

```diff
@@ -0,0 +1,454 @@
+	display_name = "Camera Abstract Type"
+	desc = "This is the abstract parent type - do not use this directly!"
+	category = "Entity"
+	circuit_flags = CIRCUIT_NO_DUPLICATES
+
+	/// Starts the cameraa
+	var/datum/port/input/start
+	/// Stops the program.
+	var/datum/port/input/stop
+	/// Camera range flag (near/far)
+	var/datum/port/input/camera_range
+	/// The network to use
+	var/datum/port/input/network
+
+	/// Allow camera range to be set or not
+	var/camera_range_settable = 1
+
+	/// Camera object
+	var/obj/machinery/camera/shell_camera = null
+	/// The shell storing the parent circuit
+	var/atom/movable/shell_parent = null
+	/// Camera random ID
+	var/c_tag_random = 0
+
+	/// Used to store the current process state
+	var/current_camera_state = FALSE
+	/// Used to store the camera emp state
+	var/current_camera_emp = FALSE
+	/// Used to store the camera emp timer id
+	var/current_camera_emp_timer_id
+	/// Used to store the last string used for the camera name
+	var/current_camera_name = ""
+	/// Used to store the current camera range setting (near/far)
+	var/current_camera_range = 0
+	/// Used to store the last string used for the camera network
+	var/current_camera_network = ""
+
+/obj/item/circuit_component/compare/remotecam/get_ui_notices()
+	. = ..()
+	if(camera_range_settable)
+		. += create_ui_notice("Energy Usage For Near (0) Range: [display_energy(REMOTECAM_ENERGY_USAGE_NEAR)] Per [DisplayTimeText(COMP_CLOCK_DELAY)]", "orange", "clock")
+		. += create_ui_notice("Energy Usage For Far (1) Range: [display_energy(REMOTECAM_ENERGY_USAGE_FAR)] Per [DisplayTimeText(COMP_CLOCK_DELAY)]", "orange", "clock")
+	else
+		. += create_ui_notice("Energy Usage While Active: [display_energy(current_camera_range > 0 ? REMOTECAM_ENERGY_USAGE_FAR : REMOTECAM_ENERGY_USAGE_NEAR)] Per [DisplayTimeText(COMP_CLOCK_DELAY)]", "orange", "clock")
+
+/obj/item/circuit_component/compare/remotecam/populate_ports()
+	. = ..()
+	compare.name = "Is Active" //Rename compare port
+
+/obj/item/circuit_component/compare/remotecam/populate_custom_ports()
+	start = add_input_port("Start", PORT_TYPE_SIGNAL)
+	stop = add_input_port("Stop", PORT_TYPE_SIGNAL)
+	if(camera_range_settable)
+		camera_range = add_input_port("Camera Range", PORT_TYPE_NUMBER, default = 0)
+	network = add_input_port("Network", PORT_TYPE_STRING, default = "ss13")
+
+	if(camera_range_settable)
+		current_camera_range = camera_range.value
+	c_tag_random = rand(1, 999)
+
+/obj/item/circuit_component/compare/remotecam/register_shell(atom/movable/shell)
+	shell_parent = shell
+	stop_process()
+	. = ..()
+
+/obj/item/circuit_component/compare/remotecam/unregister_shell(atom/movable/shell)
+	stop_process()
+	remove_camera()
+	. = ..()
+
+/obj/item/circuit_component/compare/remotecam/Destroy()
+	stop_process()
+	remove_camera()
+	current_camera_state = FALSE
+	return ..()
+
+/obj/item/circuit_component/compare/remotecam/do_comparisons()
+	return shell_camera ? current_camera_state : FALSE
```

Comment:
You'll want to just update this dynamically.
Whenever the current_camera_state is updated, update a port that dictates whether it is on or off.

Most components have already made the commitment to stick to the 'provide updates dynamically' model and it'll introduce inconsistency to change that now.
I can see why you'd do it this way, but I think comparison components should largely exist as a simple abstraction rather than integrated within action components.

Additionally, since `current_camera_state` is completely handled by this component and the inputs it receives, this comparison aspect is pretty redundant since a user should know, barring that the shell is not of the correct type, whether the camera is currently active or not.

If they really want to guarantee a camera is on or not, a simple number output should suffice.